### ISSUE TITLE
pip version of 1.9.3 beta has incorrect attr names in extract.py

### ISSUE THREAD

**Issue by BrianMWest** [IBMStreams/streamsx.topology#1669]
Playing around more with the 1.9.3 beta. Installed to Python 3.6.5 via:

    pip3.6 install streamsx==1.9.3b

I noticed that the extract was not correctly identifying functions that I was using. It checked for the presence of the attribute `_splpy_optype` and found that it was `None`, so it ignored the operator.

I checked the attributed on the function and found that there was an `__splpy_optype` with 2 underscores. I found similar issues throughout the `extract.py` file. After changing these, it identified the operator correctly, though it has some other issues.

The locations are:

    def _optype
    def _opfile
    def _opcallable
    def _opdoc
    def _create_op_parameters
        # in _splpy_wrapped
    def _create_ip_spldoc
        # opobj._splpy_style
    def _create_primitive
        # cls._splpy_input_ports
    def _write_style_info
        # _splpy_style
        # _splpy_fixed_count

That said, I found that modifying `_splpy_style` in `_create_ip_spldoc` and `_write_style_info` does not work. Leaving it alone says that it cannot find the attribute `_splpy_style`, but adding the extra underscore causes it to be unable to find: `_Extractor__splpy_style`. I checked the attributes available with `dir(opobj)` and found that there _is_ an attribute called `__splpy_style` (with 2 underscores). I confess that I don't know enough about Python to know why it is adding the extra `_Extractor` prior to the `__splpy_style` when I attempt to have 2 underscores.

I removed the body of those functions the command runs successfully, but I can't compile an graph using the operator because I get `"Undefined subroutine &TestOperator_cpp::splpy_ParamStyle called"`

I'm assuming this is because I removed the bodies of `_write_style_info` and `_create_ip_spldoc`. Unfortunately, I'm stuck at this point. I'm not sure where to go from here to further diagnose the issue.

In summary, without making the above changes, `spl-python-extract -i .` doesn't see the operator. Making the changes, the script finds the operator and generates the `cgt/xml` files, but it doesn't compile.

**Comment by BrianMWest** [comment]
I got around the `__splpy_style` issue by changing it from `opobj.__splp_style` to `getattr(opobj, "__splpy_style")`

I notice that the function:

    @spl.map
    def TestOperator(**t):
        return {"B": "Hello, world!"}

has the same issue with `__splpy_fixed_count`. In this case, though, `__splpy_fixed_count` truly does not exist in the attributes, so the same fix doesn't work.

**Comment by BrianMWest** [comment]
Without changing anything else, I switched from the `pip` version of the toolkit to the tarball on the Releases page. The tarball does not have the same issue.

**Comment by ddebrunner** [comment]
Strange - I'll take a look.

**Comment by ddebrunner** [comment]
@BrianMWest Can you provide a complete example of what is failing for you? I tried with your `TestOperator` from above and it worked.

`opt/python/streams/to.py`

```
from streamsx.spl import spl

def spl_namespace():
    return 'myops'

@spl.map()
def TestOperator(**t):
    return {"B": "Hello, World"}
```

`myapp/main.spl`

```
namespace myapp;

public composite Main {
    graph
      stream<rstring B> A = Beacon() { param iterations: 1;}
      stream<A> P = myops::TestOperator(A) {}
      () as E = Custom(P) {logic onTuple P: println(P);}
}
```

Extract the operators:
```
[streamsadmin@streamsqse tk_1669]$ spl-python-extract -i .
Checking  ./opt/python/streams/to.py for operators
myops::TestOperator
Copy globalization resources zh_CN
Copy globalization resources ja_JP
Copy globalization resources pt_BR
Copy globalization resources es_ES
Copy globalization resources it_IT
Copy globalization resources en_US
Copy globalization resources ru_RU
Copy globalization resources de_DE
Copy globalization resources fr_FR
Copy globalization resources ko_KR
Copy globalization resources zh_TW
Check info.xml: ./info.xml
WARNING: File info.xml not found. Creating info.xml from template
```

Compile and run the app:
```
[streamsadmin@streamsqse tk_1669]$ sc -M myapp::Main
Checking the constraints.
Creating the types.
Creating the functions.
Creating the operators.
Creating the standalone application.
Creating the application model.
Building the binaries.
 [CXX-type] tuple<rstring B>
 [CXX-operator] A
 [CXX-operator] P
 [CXX-operator] E
 [LD-so] so A
 [LD-so] so E
 [CXX-standalone] standalone
 [LD-standalone] standalone
 [LN-standalone] standalone 
 [LD-so] so P
 [Bundle] myapp.Main.sab
[streamsadmin@streamsqse tk_1669]$ output/bin/standalone
{B="Hello, World"}
```

**Comment by ddebrunner** [comment]
Note I am running with the fix for #1668

**Comment by BrianMWest** [comment]
@ddebrunner What version of Python are you using?

I have Python 3.6.5 installed. I removed the streamsx package and reinstalled the 1.9.3b version, then I incorporated the change to fix #1668. It runs successfully, but the issue is that it just doesn't seem to see the spl annotation in the Python script, so it doesn't generate any operator models for it.

**Comment by BrianMWest** [comment]
Here is a complete example:

`./Test.spl`

    use me.bwest.streams.test::*;
    
    composite Test {
        graph
            stream<rstring message> messages = Beacon() {
                param
                    period: 1.0;
                output messages: message="Hello, world!";

            () as sink = TestSink(messages) {
            }
    }

`./opt/python/streams/Test.py`

    from streamsx.spl import spl


    def splNamespace():
        return "me.bwest.streams.test"


    @spl.for_each()
    def TestSink(**tuple_):
        print(tuple_)

Then I run:

`python3.6 -m streamsx.scripts.extract -i . -v`

    Checking  ./opt/python/streams/Test.py for operators
    Copy globalization resources zh_CN
    Written: ./impl/nl/zh_CN/TopologySplpyResource.xlf
    Copy globalization resources ja_JP
    Written: ./impl/nl/ja_JP/TopologySplpyResource.xlf
    Copy globalization resources pt_BR
    Written: ./impl/nl/pt_BR/TopologySplpyResource.xlf
    Copy globalization resources es_ES
    Written: ./impl/nl/es_ES/TopologySplpyResource.xlf
    Copy globalization resources it_IT
    Written: ./impl/nl/it_IT/TopologySplpyResource.xlf
    Copy globalization resources en_US
    Written: ./impl/nl/en_US/TopologySplpyResource.xlf
    Copy globalization resources ru_RU
    Written: ./impl/nl/ru_RU/TopologySplpyResource.xlf
    Copy globalization resources de_DE
    Written: ./impl/nl/de_DE/TopologySplpyResource.xlf
    Copy globalization resources fr_FR
    Written: ./impl/nl/fr_FR/TopologySplpyResource.xlf
    Copy globalization resources ko_KR
    Written: ./impl/nl/ko_KR/TopologySplpyResource.xlf
    Copy globalization resources zh_TW
    Written: ./impl/nl/zh_TW/TopologySplpyResource.xlf
    Available languages for TopologySplpy resource: ['zh_CN', 'ja_JP', 'pt_BR', 'es_ES', 'it_IT', 'en_US', 'ru_RU', 'de_DE', 'fr_FR', 'ko_KR', 'zh_TW']
    Check info.xml: ./info.xml
    WARNING: File info.xml not found. Creating info.xml from template

`ls`

    impl
    info.xml
    opt
    Test.spl

I don't get any generated operator models. If I run the version from the tarball, then I do at this point. It also has an added message

**Comment by ddebrunner** [comment]
I'm running 3.5, but will check with 3.6

**Comment by ddebrunner** [comment]
@BrianMWest 1.9.5 has been released as a `streamsx` package, can you try with that version. Thanks!
